jsonrpc/filter_manager_test.go

```diff
@@ -178,6 +178,29 @@ func TestFilterWebsocket(t *testing.T) {
 	}
 }
 
+// 	This test verifies that websocket subscriptions
+//	are not susceptible to timeouts (like filters).
+func TestWSResponsiveness(t *testing.T) {
+	store := newMockStore()
+	fm := NewFilterManager(hclog.NewNullLogger(), store)
+
+	//	set the timeout to 1s, so it starts early
+	fm.timeout = 1 * time.Second
+
+	//	create block filter (as subscription)
+	mockWS := &mockWsConn{msgCh: make(chan []byte, 1)}
+	filterID := fm.NewBlockFilter(mockWS)
+
+	//	start the manager
+	go fm.Run()
+
+	//	wait for timeout to
+	time.Sleep(2 * time.Second)
```

Comment:
I don't think this is a great way to test this functionality.

You are in complete control of the testing environment here, and you know the implementation will never set up a timeout for you, but will just handle requests if it's a WS connection.

If you want to add this test, if at all, I think you could probably refactor it to use the `WriteMessage` method on `mockWs`, just send out a huge batch of messages and make sure they get delivered 

cc @Kourin1996

Reply:
The original test was written to confirm the filter (websocket) was getting uninstalled after a timeout (which is why I set it to 1 second in the test). This test was meant to confirm there **is** a problem. 

So after the fix, how do you verify that the problem is gone ? Conceptually, your batch solution is no different than the existing [TestFilterWebscoket](https://github.com/0xPolygon/polygon-edge/blob/cd02189040f33ed77f68c54d72a50bc479f061ca/jsonrpc/filter_manager_test.go#L147) - it's just a difference of 1 vs 100 responses sent. 

In that case, we should just increase the payload in that test and remove the new one altogether.

Reply:
That sounds like a plan to me. I would honestly be ok with dropping this test entirely 👍

Reply:
[Removed](https://github.com/0xPolygon/polygon-edge/pull/417/commits/354d1cabaebe78aa5615808db28328f9fede66ac)